backend/src/main/java/gov/cdc/usds/simplereport/service/DeviceTypeService.java

```diff
@@ -83,72 +88,85 @@ public DeviceSpecimenType getDefaultForDeviceId(UUID deviceId) {
-    if (st.isDeleted()) {
-      throw new IllegalGraphqlArgumentException("swab type has been deleted and cannot be used");
+    if (updateDevice.getSwabTypes() != null) {
+      List<SpecimenType> updatedSpecimenTypes =
+          updateDevice.getSwabTypes().stream()
+              .map(uuid -> _specimenTypeRepo.findById(uuid))
+              .filter(Optional::isPresent)
+              .map(Optional::get)
+              .collect(Collectors.toList());
+
+      updatedSpecimenTypes.forEach(
+          specimenType -> {
+            if (specimenType.isDeleted()) {
+              throw new IllegalGraphqlArgumentException(SWAB_TYPE_DELETED_MESSAGE);
+            }
+          });
+
+      List<DeviceSpecimenType> newDeviceSpecimenTypes =
+          updatedSpecimenTypes.stream()
+              .map(specimenType -> new DeviceSpecimenType(device, specimenType))
+              .collect(Collectors.toList());
+
+      List<DeviceSpecimenType> exitingDeviceSpecimenTypes =
+          _deviceSpecimenRepo.findAllByDeviceType(device);
+
+      // delete old ones
+      ArrayList<DeviceSpecimenType> toBeDeletedDeviceSpecimenTypes =
+          new ArrayList<>(exitingDeviceSpecimenTypes);
+      toBeDeletedDeviceSpecimenTypes.removeAll(newDeviceSpecimenTypes);
+      _deviceSpecimenRepo.deleteAll(toBeDeletedDeviceSpecimenTypes);
+
+      // create new ones
+      ArrayList<DeviceSpecimenType> toBeAddedDeviceSpecimenTypes =
+          new ArrayList<>(newDeviceSpecimenTypes);
+      toBeAddedDeviceSpecimenTypes.removeAll(exitingDeviceSpecimenTypes);
+      _deviceSpecimenRepo.saveAll(toBeAddedDeviceSpecimenTypes);
     }
-    DeviceType dt =
-        _repo.save(
-            new DeviceType(
-                name, manufacturer, model, loincCode, swabType, determineTestLength(name)));
-    _deviceSpecimenRepo.save(new DeviceSpecimenType(dt, st));
-    return dt;
+    return _repo.save(device);
   }
 
   @Transactional(readOnly = false)
   @AuthorizationConfiguration.RequireGlobalAdminUser
-  public DeviceType createDeviceTypeNew(
-      String name, String model, String manufacturer, String loincCode, List<UUID> swabTypes) {
+  public DeviceType createDeviceType(CreateDeviceType createDevice) {
 
     List<SpecimenType> specimenTypes =
-        swabTypes.stream()
+        createDevice.getSwabTypes().stream()
             .map(uuid -> _specimenTypeRepo.findById(uuid).get())
             .collect(Collectors.toList());
 
     specimenTypes.forEach(
         specimenType -> {
           if (specimenType.isDeleted()) {
-            throw new IllegalGraphqlArgumentException(
-                "swab type has been deleted and cannot be used");
+            throw new IllegalGraphqlArgumentException(SWAB_TYPE_DELETED_MESSAGE);
           }
         });
 
     DeviceType dt =
         _repo.save(
-            new DeviceType(name, manufacturer, model, loincCode, null, determineTestLength(name)));
+            new DeviceType(
+                createDevice.getName(),
```

Comment:
Just a nit, but could you add a constructor to `DeviceType` that takes a `CreateDeviceType` as the input?